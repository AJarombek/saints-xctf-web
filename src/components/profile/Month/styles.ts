/**
 * JSS styles for the Month component.
 * @author Andrew Jarombek
 * @since 10/19/2020
 */

export default {
  month: {
    border: '1px solid #555',
  },
};
