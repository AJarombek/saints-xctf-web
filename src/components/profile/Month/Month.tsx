/**
 * Component for a month on the calendar.
 * @author Andrew Jarombek
 * @since 10/19/2020
 */

import React, {useMemo} from 'react';
import {createUseStyles} from "react-jss";
import styles from "./styles";
import Week from "../Week";
import {RangeViewItemsMeta} from "../../../redux/types";
import moment from "moment";

interface IProps {
    rangeView: RangeViewItemsMeta;
    start: moment.Moment;
    monthStart: moment.Moment;
    monthEnd: moment.Moment;
}

const useStyles = createUseStyles(styles);

const Month: React.FunctionComponent<IProps> = ({ rangeView, start, monthStart, monthEnd }) => {
    const classes = useStyles();

    const weeks = useMemo(() => {
        let startDate = start.clone();
        let endDate = start.clone().add(6, 'days');
        const rangeViewItems = rangeView?.items ?
            rangeView.items.map((item) => ({ ...item, date: moment(item.date) })) : [];

        // The first item in weekList is the start date of the week.  The second item in weekList is the range view
        // for the week.
        const weekList = [] as any[];
        for (let i = 0; i < 6; i++) {
            const items = [];
            while (rangeViewItems.length && rangeViewItems[0].date >= startDate && rangeViewItems[0].date <= endDate) {
                items.push(rangeViewItems.shift());
            }

            weekList.push([startDate.clone(), items]);
            startDate.add(1, 'week');
            endDate.add(1, 'week');
        }

        console.info(weekList)
        return weekList;
    }, [start, rangeView]);

    return (
        <div className={classes.month}>
            {weeks.map(([weekStart, rangeViewItems]) => (
                <Week start={weekStart} monthStart={monthStart} monthEnd={monthEnd} rangeViewItems={rangeViewItems} />
            ))}
        </div>
    );
};

export default Month;
