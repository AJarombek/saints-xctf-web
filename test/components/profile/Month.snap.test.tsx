/**
 * Snapshot test for the Month component.
 * @author Andrew Jarombek
 * @since 3/6/2021
 */

import React from 'react';
import renderer from 'react-test-renderer';
import moment from 'moment';
import Month from '../../../src/components/profile/Month/Month';

describe('Month Snapshot Tests', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Month
          rangeView={{
            items: [
              {
                date: '2021-03-02',
                feel: 5,
                miles: 4.5
              },
              {
                date: '2021-03-03',
                feel: 5,
                miles: 5.45
              },
              {
                date: '2021-03-04',
                feel: 5,
                miles: 5.45
              },
              {
                date: '2021-03-05',
                feel: 7,
                miles: 6.3
              },
              {
                date: '2021-03-06',
                feel: 6,
                miles: 6.01
              }
            ]
          }}
          start={moment('2021-03-01')}
          monthStart={moment('2021-03-01')}
          monthEnd={moment('2021-03-31')}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
