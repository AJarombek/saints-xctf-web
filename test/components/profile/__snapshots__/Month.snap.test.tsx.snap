// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Month Snapshot Tests renders correctly 1`] = `
<div
  className="month-0-1-1"
>
  <div
    className="week-0-1-2"
  >
    <div
      className="day-0-1-3 currentMonth-0-1-5"
    >
      <div>
        <p
          className="dayOfMonth-0-1-6"
        >
          1
        </p>
      </div>
    </div>
    <div
      className="day-0-1-3 currentMonth-0-1-5 feel-0-1-4 feel-d1-0-1-9"
    >
      <div>
        <p
          className="dayOfMonth-0-1-6"
        >
          2
        </p>
        <div
          className="miles-0-1-7"
        >
          <p>
            4.50
          </p>
          <p>
            Miles
          </p>
        </div>
      </div>
    </div>
    <div
      className="day-0-1-3 currentMonth-0-1-5 feel-0-1-4 feel-d2-0-1-10"
    >
      <div>
        <p
          className="dayOfMonth-0-1-6"
        >
          3
        </p>
        <div
          className="miles-0-1-7"
        >
          <p>
            5.45
          </p>
          <p>
            Miles
          </p>
        </div>
      </div>
    </div>
    <div
      className="day-0-1-3 currentMonth-0-1-5 feel-0-1-4 feel-d3-0-1-11"
    >
      <div>
        <p
          className="dayOfMonth-0-1-6"
        >
          4
        </p>
        <div
          className="miles-0-1-7"
        >
          <p>
            5.45
          </p>
          <p>
            Miles
          </p>
        </div>
      </div>
    </div>
    <div
      className="day-0-1-3 currentMonth-0-1-5 feel-0-1-4 feel-d4-0-1-12"
    >
      <div>
        <p
          className="dayOfMonth-0-1-6"
        >
          5
        </p>
        <div
          className="miles-0-1-7"
        >
          <p>
            6.30
          </p>
          <p>
            Miles
          </p>
        </div>
      </div>
    </div>
    <div
      className="day-0-1-3 currentMonth-0-1-5 feel-0-1-4 feel-d5-0-1-13"
    >
      <div>
        <p
          className="dayOfMonth-0-1-6"
        >
          6
        </p>
        <div
          className="miles-0-1-7"
        >
          <p>
            6.01
          </p>
          <p>
            Miles
          </p>
        </div>
      </div>
    </div>
    <div
      className="day-0-1-3 currentMonth-0-1-5"
    >
      <div>
        <p
          className="dayOfMonth-0-1-6"
        >
          7
        </p>
      </div>
    </div>
    <div
      className="weekTotal-0-1-15"
    >
      <div
        className="miles-0-1-16"
      >
        <p>
          27.71
        </p>
        <p>
          Miles
        </p>
      </div>
    </div>
  </div>
  <div
    className="week-0-1-2"
  >
    <div
      className="day-0-1-3 currentMonth-0-1-5"
    >
      <div>
        <p
          className="dayOfMonth-0-1-6"
        >
          8
        </p>
      </div>
    </div>
    <div
      className="day-0-1-3 currentMonth-0-1-5"
    >
      <div>
        <p
          className="dayOfMonth-0-1-6"
        >
          9
        </p>
      </div>
    </div>
    <div
      className="day-0-1-3 currentMonth-0-1-5"
    >
      <div>
        <p
          className="dayOfMonth-0-1-6"
        >
          10
        </p>
      </div>
    </div>
    <div
      className="day-0-1-3 currentMonth-0-1-5"
    >
      <div>
        <p
          className="dayOfMonth-0-1-6"
        >
          11
        </p>
      </div>
    </div>
    <div
      className="day-0-1-3 currentMonth-0-1-5"
    >
      <div>
        <p
          className="dayOfMonth-0-1-6"
        >
          12
        </p>
      </div>
    </div>
    <div
      className="day-0-1-3 currentMonth-0-1-5"
    >
      <div>
        <p
          className="dayOfMonth-0-1-6"
        >
          13
        </p>
      </div>
    </div>
    <div
      className="day-0-1-3 currentMonth-0-1-5"
    >
      <div>
        <p
          className="dayOfMonth-0-1-6"
        >
          14
        </p>
      </div>
    </div>
    <div
      className="weekTotal-0-1-15"
    >
      <div
        className="miles-0-1-16"
      >
        <p>
          0.00
        </p>
        <p>
          Miles
        </p>
      </div>
    </div>
  </div>
  <div
    className="week-0-1-2"
  >
    <div
      className="day-0-1-3 currentMonth-0-1-5"
    >
      <div>
        <p
          className="dayOfMonth-0-1-6"
        >
          15
        </p>
      </div>
    </div>
    <div
      className="day-0-1-3 currentMonth-0-1-5"
    >
      <div>
        <p
          className="dayOfMonth-0-1-6"
        >
          16
        </p>
      </div>
    </div>
    <div
      className="day-0-1-3 currentMonth-0-1-5"
    >
      <div>
        <p
          className="dayOfMonth-0-1-6"
        >
          17
        </p>
      </div>
    </div>
    <div
      className="day-0-1-3 currentMonth-0-1-5"
    >
      <div>
        <p
          className="dayOfMonth-0-1-6"
        >
          18
        </p>
      </div>
    </div>
    <div
      className="day-0-1-3 currentMonth-0-1-5"
    >
      <div>
        <p
          className="dayOfMonth-0-1-6"
        >
          19
        </p>
      </div>
    </div>
    <div
      className="day-0-1-3 currentMonth-0-1-5"
    >
      <div>
        <p
          className="dayOfMonth-0-1-6"
        >
          20
        </p>
      </div>
    </div>
    <div
      className="day-0-1-3 currentMonth-0-1-5"
    >
      <div>
        <p
          className="dayOfMonth-0-1-6"
        >
          21
        </p>
      </div>
    </div>
    <div
      className="weekTotal-0-1-15"
    >
      <div
        className="miles-0-1-16"
      >
        <p>
          0.00
        </p>
        <p>
          Miles
        </p>
      </div>
    </div>
  </div>
  <div
    className="week-0-1-2"
  >
    <div
      className="day-0-1-3 currentMonth-0-1-5"
    >
      <div>
        <p
          className="dayOfMonth-0-1-6"
        >
          22
        </p>
      </div>
    </div>
    <div
      className="day-0-1-3 currentMonth-0-1-5"
    >
      <div>
        <p
          className="dayOfMonth-0-1-6"
        >
          23
        </p>
      </div>
    </div>
    <div
      className="day-0-1-3 currentMonth-0-1-5"
    >
      <div>
        <p
          className="dayOfMonth-0-1-6"
        >
          24
        </p>
      </div>
    </div>
    <div
      className="day-0-1-3 currentMonth-0-1-5"
    >
      <div>
        <p
          className="dayOfMonth-0-1-6"
        >
          25
        </p>
      </div>
    </div>
    <div
      className="day-0-1-3 currentMonth-0-1-5"
    >
      <div>
        <p
          className="dayOfMonth-0-1-6"
        >
          26
        </p>
      </div>
    </div>
    <div
      className="day-0-1-3 currentMonth-0-1-5"
    >
      <div>
        <p
          className="dayOfMonth-0-1-6"
        >
          27
        </p>
      </div>
    </div>
    <div
      className="day-0-1-3 currentMonth-0-1-5"
    >
      <div>
        <p
          className="dayOfMonth-0-1-6"
        >
          28
        </p>
      </div>
    </div>
    <div
      className="weekTotal-0-1-15"
    >
      <div
        className="miles-0-1-16"
      >
        <p>
          0.00
        </p>
        <p>
          Miles
        </p>
      </div>
    </div>
  </div>
  <div
    className="week-0-1-2"
  >
    <div
      className="day-0-1-3 currentMonth-0-1-5"
    >
      <div>
        <p
          className="dayOfMonth-0-1-6"
        >
          29
        </p>
      </div>
    </div>
    <div
      className="day-0-1-3 currentMonth-0-1-5"
    >
      <div>
        <p
          className="dayOfMonth-0-1-6"
        >
          30
        </p>
      </div>
    </div>
    <div
      className="day-0-1-3 currentMonth-0-1-5"
    >
      <div>
        <p
          className="dayOfMonth-0-1-6"
        >
          31
        </p>
      </div>
    </div>
    <div
      className="day-0-1-3"
    >
      <div>
        <p
          className="dayOfMonth-0-1-6"
        >
          1
        </p>
      </div>
    </div>
    <div
      className="day-0-1-3"
    >
      <div>
        <p
          className="dayOfMonth-0-1-6"
        >
          2
        </p>
      </div>
    </div>
    <div
      className="day-0-1-3"
    >
      <div>
        <p
          className="dayOfMonth-0-1-6"
        >
          3
        </p>
      </div>
    </div>
    <div
      className="day-0-1-3"
    >
      <div>
        <p
          className="dayOfMonth-0-1-6"
        >
          4
        </p>
      </div>
    </div>
    <div
      className="weekTotal-0-1-15"
    >
      <div
        className="miles-0-1-16"
      >
        <p>
          0.00
        </p>
        <p>
          Miles
        </p>
      </div>
    </div>
  </div>
  <div
    className="week-0-1-2"
  >
    <div
      className="day-0-1-3"
    >
      <div>
        <p
          className="dayOfMonth-0-1-6"
        >
          5
        </p>
      </div>
    </div>
    <div
      className="day-0-1-3"
    >
      <div>
        <p
          className="dayOfMonth-0-1-6"
        >
          6
        </p>
      </div>
    </div>
    <div
      className="day-0-1-3"
    >
      <div>
        <p
          className="dayOfMonth-0-1-6"
        >
          7
        </p>
      </div>
    </div>
    <div
      className="day-0-1-3"
    >
      <div>
        <p
          className="dayOfMonth-0-1-6"
        >
          8
        </p>
      </div>
    </div>
    <div
      className="day-0-1-3"
    >
      <div>
        <p
          className="dayOfMonth-0-1-6"
        >
          9
        </p>
      </div>
    </div>
    <div
      className="day-0-1-3"
    >
      <div>
        <p
          className="dayOfMonth-0-1-6"
        >
          10
        </p>
      </div>
    </div>
    <div
      className="day-0-1-3"
    >
      <div>
        <p
          className="dayOfMonth-0-1-6"
        >
          11
        </p>
      </div>
    </div>
    <div
      className="weekTotal-0-1-15"
    >
      <div
        className="miles-0-1-16"
      >
        <p>
          0.00
        </p>
        <p>
          Miles
        </p>
      </div>
    </div>
  </div>
</div>
`;
